### PR TITLE
new features for `CESMCollections`

### PR DESCRIPTION
**Pull request by matt-long**
This includes some new features:
- `cache_directory` is settable via `set_options` and searchable during `build_collection`
- added "input-file" option for `StorageResource`, enabling minimal test for `build_collection`
- reformatted input files to put `locations` at the level of `experiment`, avoiding extensive duplication
- included `direct_access` and `resource_type` attributes in database

### CODE REVIEW

**Comment by matt-long** [comment]
Forgot to mention: this closes #19. 

It furthers progress toward #10.

**Comment by matt-long** [comment]
@andersy005, I just pushed another change that occurred to me as I started to use this version.

**Comment by andersy005** [comment]
This looks good to me! one quick question: I 've noticed that `cesm_definitions.yml` config file is still part of the package. Moving forward, are we going to keep it as part of the package?

**Comment by andersy005** [comment]
@matt-long, this is a request :): When you get time, can you update the documentation to reflect the new features you added recently (yesterday and today)? Let me know If you have so much on your plate, and I will work on it.

File: intake_cesm/manage_collections.py [hunk 1]
```diff
@@ -73,11 +78,22 @@ def _list_files_hsi(self):
 
         return filelist
 
+    def _list_files_input_file(self):
+        with open(self.urlpath, "r") as fid:
+            return fid.read().splitlines()
+
 
 class CESMCollections(object):
```
Comment (andersy005) [inline intake_cesm/manage_collections.py]: Docstrings for the class constructor and some public methods would be useful to have.

**Comment by matt-long** [comment]
@andersy005 thanks for the reminder. I'll try to get to the docs, but would appreciate your help as I am strapped for time.

**Comment by andersy005** [comment]
Sure.. I will start working on the documentation this afternoon. I am about to merge this PR. Once again, thank you!

File: intake_cesm/manage_collections.py (hunk 1 above)
Comment (matt-long) [reply intake_cesm/manage_collections.py]: fair enough. I added some minimal text, but will need to go back thru carefully.  Still sketching out functionality...and working on this is a guilty pleasure (I should really be doing about 100 other things.

Comment (andersy005) [reply intake_cesm/manage_collections.py]: I appreciate your work! Don't worry about it. I will take care of it in a separate PR that way you have time to do other things as well.

**Pull request merged**